protocol/src/main/java/org/geysermc/mcprotocollib/protocol/MinecraftProtocol.java

```diff
@@ -84,7 +90,8 @@ public MinecraftProtocol(PacketCodec codec) {
         this.codec = codec;
         this.targetState = ProtocolState.STATUS;
 
-        this.setState(ProtocolState.HANDSHAKE);
+        this.setInboundState(ProtocolState.HANDSHAKE);
```

Comment:
Seems like we could just have a setState/setStates method just for convenience.

Reply:
Is this really necessary? The only real reason for the protocol to be set for both at once is for handshake. There's currently no other reason for setting both inbound and outbound at once. There are also separate operations on the Session used for switching protocols.

Reply:
It's done three times here. Might as well have a private method for it?

Reply:
We could have a private method to reset current state to handshake. Will add that tomorrow.

Reply:
done